rfcs/experimental/flow_graph_wait_for_one/README.md

```diff
@@ -0,0 +1,481 @@
+the postponed message and the corresponding metainfo from the queue and spawns a task to process it.
+
+Since the ``function_node`` guarantees that all of the elements would be retrieved from the internal queue at some time, [buffering issues](#buffering-the-metainfo) cannot take place.
+
+### Rejecting ``function_node``
+
+If the concurrency of the node is set to ``unlimited``, behaves the same as in the ``queueing`` case described above.
+
+Otherwise, if the concurrency limit of the node is reached, both message and the associated metainformations would be rejected and the predecessor that called the ``try_put_task``
+is responsible on buffering both of them.
+
+If the predecessor is not the buffering node, both message and the metainfo would be lost.
+
+When some ``function_node`` task is completed, it will try to get a buffered message and the metainfo from the predecessor by calling the ``try_get(msg, metainfo)`` method. 
+
+Since the ``function_node`` guarantees that all of the elements would be retrieved from the predecessor, [buffering issues](#buffering-the-metainfo) cannot take place
+for buffering nodes, preceding the ``function_node``.
+
+### Lightweight ``function_node``
+
+Calls to ``try_put_task`` in the lightweight node will operate on the concurrency limit of the node in the same manner as is defined by the message buffering policy -
+``queueing`` (default)  or ``rejecting``.
+
+The difference is that for lightweight nodes the tasks would not be spawned in most of the cases and the node body will be executed by the calling thread. 
+Since there are no tasks, the calling thread would broadcast the output and the metainformation to the successors after completing the function.
+
+### ``continue_node``
+
+``continue_node`` only executes the associated body (and broadcasts the signal to the successors) when it receives ``N`` signals from the predecessors, where ``N``
+is the number of predecessors. 
+
+It means that prior to executing the body, the node can receive several ``metainfo`` instances from different predecessors. To handle this, the node initially stores an
+empty metainfo instance inside itself and each call to ``try_put_task`` with non-empty metainfo, merges the received metainformation with the stored instance.
+
+Additional reference counter would be held on each input ``wait_context`` to make sure the corresponding ``try_put_and_wait`` will remain blocked until the item would leave the
+``continue_node``. 
+
+When the ``continue_node`` receives the last signal from the predecessors, it creates a task to complete the associated body. Once this task is executed, the metainfo instance stored in
+the ``continue_node`` switches back to the empty state for further work and the previously accumulated metainfo is broadcasted to the successors of the node.
```

Comment:
As described, there seems to be a race in case the `continue_node` can be triggered more than once during an execution of the graph. In such case, new messages might come while the body of the node is executed - so switching the state back to empty should seemingly happen before executing the body.

Reply:
Thanks, it is an incorrect description. The implementation resets the stored metainfo before spawning the task under the node's mutex. 
Updated the description.